numpy/ma/core.py

```diff
@@ -93,6 +93,33 @@
 class MaskedArrayFutureWarning(FutureWarning):
     pass
 
+def _deprecate_argsort_axis(arr, axis):
+    """
+    Takes the array argsort was called upon, and the axis argument
+    it was called with.
+
+    np.ma.argsort has a long-term bug where the default of the axis argument
+    is wrong (gh-8701), which now must be kept for backwards compatibiity.
+    Thankfully, this only makes a difference when arrays are 2- or more-
+    dimensional, so we only need a warning then.
+    """
+    if axis is np._NoValue:
+        if arr.ndim <= 1:
+            # no warning needed - switch to -1 to avoid surprising subclasses
+            return -1
+        else:
+            # 2017-04-10, numpy 1.13.0
+            warnings.warn(
+                "Unlike np.argsort, np.sort, np.ma.sort, and the "
```

Comment:
The benefit of having a `MaskedArrayFutureWarning` is that it makes it obvious that the warning only applies to masked arrays, and the message doesn't need to explain that only `ma.argsort` has a bad default argument